wgengine/userspace.go

```diff
@@ -187,6 +187,11 @@ func NewUserspaceEngine(logf logger.Logf, tunName string, listenPort uint16) (En
 	}
 	logf("CreateTUN ok.")
 
+	if err := waitInterfaceUp(tun, 90*time.Second, logf); err != nil {
+		// TODO: Can we ignore this error and simply proceed?
```

Comment:
You tell us! :) You did the research. What happens if it's not up and we proceed?

Reply:
> What happens if it's not up and we proceed?

The same thing as before: currently it'll run into your patch that repeatedly attempts...

> You tell us!

I would suggest not to proceed because it cannot work if the iface isn't up. Maybe only increase the timeout to a larger value (i.e. 5 minutes).